proxy/server/server.go

```diff
@@ -270,30 +282,86 @@ func (s *T) newTLSCfg(hostCfgs map[engine.HostKey]engine.Host) (*tls.Config, err
 
 	defaultHostName := ""
 	pairs := map[string]tls.Certificate{}
+	getCertFuncs := map[string]getCertificateFunc{}
+
 	for _, hostCfg := range hostCfgs {
 		if hostCfg.Settings.Default {
 			defaultHostName = hostCfg.Name
 		}
-		c := hostCfg.Settings.KeyPair
-		if c == nil {
-			continue
-		}
-		keyPair, err := tls.X509KeyPair(c.Cert, c.Key)
-		if err != nil {
-			return nil, err
-		}
-		if hostCfg.Settings.OCSP.Enabled {
-			log.Infof("%v OCSP is enabled for %v, resolvers: %v", s, hostCfg, hostCfg.Settings.OCSP.Responders)
-			r, err := s.stapler.StapleHost(&hostCfg)
+		if hostCfg.Settings.KeyPair != nil {
+			c := hostCfg.Settings.KeyPair
+			if hostCfg.Settings.AutoCert != nil {
+				log.Warningf("Host %v has a KeyPair and AutoCert enabled. KeyPair takes precedence. Autocert generation disabled.", hostCfg.Name)
+			}
+			keyPair, err := tls.X509KeyPair(c.Cert, c.Key)
 			if err != nil {
-				log.Warningf("%v failed to staple %v, error %v", s, hostCfg, err)
-			} else if r.Response.Status == ocsp.Good || r.Response.Status == ocsp.Revoked {
-				keyPair.OCSPStaple = r.Staple
-			} else {
-				log.Warningf("%s got undefined status from OCSP responder: %v", s, r.Response.Status)
+				return nil, err
+			}
+			if hostCfg.Settings.OCSP.Enabled {
```

Comment:
We only honor OCSP.Enabled for hosts with static certificates, I guess we can look into implementing that for AutoCert hosts in the scope of different PR.